### PR TITLE
internal/ingest: expose public ingest API.

### PR DESCRIPTION
**Pull request by mhemmings**
Exposes a small public wrapper for internal/ingest.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/251/

**Review by rogpeppe — APPROVED** [review]
LGTM, but maybe make the tests in ingest_test.go use the new exported API?

**Review by mhilton — APPROVED** [review]
LGTM

File: internal/ingest/ingest.go [hunk 1]
```diff
@@ -18,6 +19,18 @@ type ingestParams struct {
 	whitelist []WhitelistEntity
 }
 
+// IngestParams holds information about the charmstores to use for ingestion and the entities to whitelist
+type IngestParams struct {
+	// Src holds the client for the charmstore to ingest from
```
Comment (rogpeppe) [inline internal/ingest/ingest.go]: s/client for the charmstore/charmstore client/ ?

and a full stop :)

File: internal/ingest/ingest.go [hunk 2]
```diff
@@ -18,6 +19,18 @@ type ingestParams struct {
...
+	Src *csclient.Client
+
+	// Dest holds the client for the charmstore to ingest into
```
Comment (rogpeppe) [inline internal/ingest/ingest.go]: ditto

File: internal/ingest/ingest.go [hunk 3]
```diff
@@ -18,6 +19,18 @@ type ingestParams struct {
...
+	Dest *csclient.Client
+
+	// Whitelist holds a slice of entities to ingest
```
Comment (rogpeppe) [inline internal/ingest/ingest.go]: full stop

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/252/

File: internal/ingest/ingest.go [hunk 4]
```diff
@@ -113,6 +126,16 @@ type IngestStats struct {
 	Errors              []string
 }
 
+// Ingest retrieves whitelisted entities from one charmstore and adds them to another, returning statistics on
```
Comment (rogpeppe) [inline internal/ingest/ingest.go]: We don't generally care that much about line length, but it's perhaps worth keeping comment
width reasonable - I use a little shell script called cfmt that runs the comment through fmt(1).

	#!/usr/bin/env rc
	9 sed 's:^(	*)//:\1:' | 9 fmt | 9 sed 's:^(	*):\1//:'

**Review by rogpeppe — APPROVED** [review]
A few more thoughts about the comments...

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/253/

**Review by rogpeppe — APPROVED** [review]
LGTM thanks!

**Pull request merged**